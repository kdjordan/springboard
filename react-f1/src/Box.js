import React from 'react'
import './Box.css'


function Box({color,height, width, id, handleDelete}) {
    const boxData = 
        {
            id,
            height,
            width,
            color
        }
 return (
    <div className='BoxWrapper'>
        <div className='Box'  style={{height:boxData.height,  width:boxData.width, backgroundColor:boxData.color }}>
        </div>
        <div>
            <button className='BoxButton' onClick={e => handleDelete(e, boxData.id)}>X</button>
        </div>
    </div>
    
 )

}

export default Box