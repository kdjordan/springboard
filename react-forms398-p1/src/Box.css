.BoxWrapper {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    margin-top: 1vh;
}
.Box {
    display: block;
    border-radius: 15px;
    max-width: 200px;
    max-height: 200px;
}

.BoxButton {
    color: var(--color-red);
    padding: .1rem .3rem;
    background-color: transparent;
    border-radius: 50%;
    border: 1px solid var(--color-red);
    font-size: 12px;
    margin-top: 1vh;
    cursor: pointer;
    transform: scale(1);
    transition: all .4s ease;
}

.BoxButton:hover {
    transform: scale(1.1);
    background-color: rgba(255, 255,255, .5);
    border-color: rgba(255, 255,255, .5);
}