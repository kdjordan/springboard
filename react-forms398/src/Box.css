.BoxWrapper {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    margin-top: 1vh;
}
.Box {
    display: block;
    border-radius: 15px;
}

.BoxButton {
    color: red;
    padding: .3rem .6rem;
    background-color: rgb(255, 255, 255, .7);
    border-radius: 50%;
    border: 2px solid red;
    font-size: 1.2rem;
    margin-top: 1vh;
    cursor: pointer;
    transform: scale(1);
    transition: all .4s ease;
}

.BoxButton:hover {
    transform: scale(1.1);
}