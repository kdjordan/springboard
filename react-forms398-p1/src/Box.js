import React from 'react'
import './Box.css'


function Box({color,height, width, id, deleteBox}) {
    const boxData = [
        {
            id: id,
            height: height,
            width: width,
            color: color
        }
    ]
 return (
    <div className='BoxWrapper'>
    <div className='Box' style={{height:boxData[0].height,  width:boxData[0].width, backgroundColor:boxData[0].color }}></div>
        <button className='BoxButton' onClick={e => deleteBox(e, boxData[0].id)}>X</button>
    </div>
 )

}

export default Box