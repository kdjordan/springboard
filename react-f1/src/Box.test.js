import React from 'react';
import ReactDOM from 'react-dom';
import Box from './Box';

it('renders without crashing', function() {
  const div = document.createElement('div');
  ReactDOM.render(<Box />, div);
  ReactDOM.unmountComponentAtNode(div);
});
