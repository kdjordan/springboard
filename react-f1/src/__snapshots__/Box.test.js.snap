// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxWrapper"
  >
    <div
      class="Box"
    />
    <button
      class="BoxButton"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
